space_invaders.py

```diff
@@ -56,31 +62,25 @@ def __init__(self, x, y):
 
         """
         pygame.sprite.Sprite.__init__(self)
-        self.image = alien_image
-        self.rect = self.image.get_rect()
-        self.rect = self.rect.move(x-self.rect.left, y-self.rect.top) # move
-                                                    # to (x, y)
-        self.velocity = alien_velocity
+        self.rect = Alien.image.get_rect()
+        self.rect = self.rect.move(x-self.rect.left, y-self.rect.top) # move to (x, y)
 
     def update(self):
         """
         Updates the alien's coordinates.  Reverses the alien's velocity at
         random or if it hits the edges of the window.
 
         """
-        self.rect = self.rect.move(self.velocity)
+        if Alien.moving_down:
+            self.rect = self.rect.move(DOWN)
+            return
+        self.rect = self.rect.move(Alien.velocity)
         if self.rect.left < 0 or self.rect.right > WIDTH: # Alien is outside of
                                                     # window
-            self.velocity[0] = -self.velocity[0]
-            self.rect = self.rect.move(self.velocity)
-            self.rect = self.rect.move(self.velocity)
-            self.velocity[1] = ALIEN_SPEED # comment out this and following
-                            # line to not have aliens get closer
+            Alien.moving_down = True
```

Comment:
so you call alien.update() for each alien object in the fleet, and each one messes with the static variables:moving_down and velocity. In that case, these variables might as well be normal class variables instead of static.